internal/repository/repository.go

```diff
@@ -536,22 +536,15 @@ func (r *Repository) KeyName() string {
 	return r.keyName
 }
 
-// List returns a channel that yields all IDs of type t in the backend.
-func (r *Repository) List(ctx context.Context, t restic.FileType) <-chan restic.ID {
-	out := make(chan restic.ID)
-	go func() {
-		defer close(out)
-		for strID := range r.be.List(ctx, t) {
-			if id, err := restic.ParseID(strID); err == nil {
-				select {
-				case out <- id:
-				case <-ctx.Done():
-					return
-				}
-			}
+// List runs fn for all files of type t in the repo.
+func (r *Repository) List(ctx context.Context, t restic.FileType, fn func(restic.ID, int64) error) error {
```

Comment:
Why callback function gets two separate `restic.ID` and `int64` parameters, and not single `restic.FileInfo` parameter?

Reply:
`restic.FileInfo` contains `Name string`, which is fine for the backend: There, the files have names. On the "repository" abstraction level, we're only dealing with `restic.ID`s. I did not see the point in combining the two parameters into a new struct, as I did not come up with a good name for it.

Do you have an idea?

Thinking about it, it may get easier to extend should we ever decide to add a third parameter. hm.

Reply:
We could also have both, `backend.FileInfo` and `repository.FileInfo`, but that may be too confusing...

Reply:
Ah, didn't notice `restic.ID` vs `string` difference. In that case, I think two parameters are a good decision (no sense to worry about hypothetical third parameter until we actually need it, IMO).

Reply:
Agreed